src/core/simulation.jl

```diff
@@ -236,6 +234,7 @@ function _check_steps(
     for (stage_number, stage_name) in sim.sequence.order
         forecast_count = length(stage_initial_times[stage_number])
         stage = get(sim.stages, stage_name, nothing)
+        @assert !isnothing(stage)
```

Comment:
Instead of this assert you could just change the line above to `stage = sim.stages[stage_name]`.